src/foam/lib/parse/ErrorReportingPStreamFactory.java

```diff
@@ -0,0 +1,43 @@
+package foam.lib.parse;
+
+import java.util.*;
+
+import foam.core.X;
+import foam.core.XFactory;
+
+public class ErrorReportingPStreamFactory implements XFactory {
+  private final ErrorReportingPStream eps;
+  private X                           x;
+  private String                      fileName;
+  public static final List<String>    listErps_ = new ArrayList<>();
+
+  public ErrorReportingPStreamFactory(ErrorReportingPStream ErrorRps, String fileName) {
+    this.eps      = ErrorRps;
+    this.fileName = fileName;
+  }
+
+  @Override
+  public Object create(X x) {
+    this.x = x;
+    return null;
+  }
+
+  public CharSequence getMessage() {
+    String msg = eps.getMessage();
+    listErps_.add(fileName + ";" + msg);
+    x.put("epsJrl", listErps_);
+    return msg;
+  }
+
+  public PStream apply(Parser parser, ParserContext x) {
+    return eps.apply(parser, x);
+  }
+
+  public static List<String> getListerps() {
```

Comment:
```suggestion
  public static List<String> getListErps() {
```
Is this possible, to be consistent? or does it clash with generated code for listErps?